Extension/src/LanguageServer/client.ts

```diff
@@ -745,6 +751,43 @@ class DefaultClient implements Client {
         }
     }
 
+    private promptCompileCommands(params: CompileCommandsPaths) : void {
+        if (this.configuration.Configurations[this.configuration.CurrentConfiguration].compileCommands !== undefined) {
+            return;
+        }
+
+        let showCompileCommandsSelection: PersistentState<boolean> = new PersistentState<boolean>("CPP.showPromptCompileCommands", true);
+        if (!showCompileCommandsSelection.Value) {
+           return;
+        }
+
+        let message: string = "Detected compile_commands.json. Would you like to select a compile_commands.json for this configuration?";
```

Comment:
Do we want to separate the simple case (only 1 found) from the "we need more information" case so that the user only has to click once?

If there is only one `compile_commands.json` found, we could just say "Would you like to use `[path]` to configure IntelliSense?"